packages/@aws-cdk/aws-ec2/lib/security-group.ts

```diff
@@ -519,10 +525,13 @@ export class SecurityGroup extends SecurityGroupBase {
    *   strictly necessary).
    */
   private addDefaultEgressRule() {
-    if (this.allowAllOutbound) {
-      this.directEgressRules.push(ALLOW_ALL_RULE);
-    } else {
-      this.directEgressRules.push(MATCH_NO_TRAFFIC);
+    if(this.disableInlineRules) {
```

Comment:
I wonder if this entire block doesn't collapse down to

```ts
this.addEgressRule(...)
```

That would do the correct analysis of whether or not inline rules are allowed or not already.

Reply:
If we call into addEgressRule and `this.allowAllOutbound` is true it will emit a warning and return without adding any security groups.

I'm more than a little hesitant to change that, but I agree this nested if statement here doesn't really tickle my fancy. I'll consider it a bit more.

Reply:
Oh fair point. Yes, I guess it does make sense to special-case it then. Together with remembering the ID for a subobject that we might need to remove later it might just make more sense to duplicate some logic.

Reply:
I was able to collapse this a bit.